### PR TITLE
Update generate sw-toolbox.js separately to optimize performance

### PR DESCRIPTION
**Pull request by nekolr**
I don't know why, it is slower to pack the contents of `sw-toolbox` together into sw.js, but faster to load them separately.

### CODE REVIEW

File: templates/sw.tpl.js [hunk 1]
```diff
@@ -1,18 +1,4 @@
...
-
+self.importScripts("lib/sw-toolbox/sw-toolbox.js");
```
Comment (PengXing) [inline templates/sw.tpl.js:1]: `importScripts` may delay entire lifecycle, and it blocks the execution of subsequent code.

Comment (PengXing) [reply templates/sw.tpl.js:1]: Another network request also may delay entire lifecycle of service worker.

**Review by PengXing — COMMENTED** [review]
The performance of generating service worker, or  installing service worker? In either case, we should find out why.

**Comment by nekolr** [comment]
Final, i found that the performance of the two methods is same, although `importScripts` does not block the main thread, but as you said, it does send a new request, so i close this pull request.

File: templates/sw.tpl.js (hunk 1 above)
Comment (nekolr) [reply templates/sw.tpl.js:1]: its installing service worker.

**Comment by PengXing** [comment]
Yes, `importScripts` does not block the main thread, but blocks execution of service worker.

BTW: thanks for your PR.

See this documentation, https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers#Importing_scripts_and_libraries

**Pull request closed without merging**